targets/TARGET_STM/hal_tick_common.c

```diff
@@ -17,9 +17,19 @@
 
 // Overwrite default HAL functions defined as "weak"
 
+// This variable is set to 1 at the of mbed_sdk_init function.
+// The ticker_read_us function must not be called until the mbed_sdk_init is terminated.
+extern int mbed_sdk_inited;
+
 uint32_t HAL_GetTick()
 {
-    return ticker_read_us(get_us_ticker_data()) / 1000; // 1 ms tick is required for ST HAL
+    // 1 ms tick is required for ST HAL driver
+    if (mbed_sdk_inited) {
+        return (ticker_read_us(get_us_ticker_data()) / 1000);
+    }
+    else {
+        return (us_ticker_read() / 1000);
```

Comment:
With the 16 bit timer this will overflow in ~66ms. If anything in the SDK init takes longer than this it may get stuck as HAL_GetTick will never count higher than this during init. As a potential solution you could accumulate elapsed time rather than reading it directly. Something like:

```C
uint32_t new_time = us_ticker_read();
elapsed_time += (new_time - prev_time) & 0xFFFF; // Only use the lower 16 bits
prev_time = new_time;
return elapsed_time / 1000;
```

Note - You'll need to make sure `prev_time` is reset in `HAL_InitTick` since this resets the timer count.

Reply:
Done in commit https://github.com/ARMmbed/mbed-os/pull/7352/commits/f785c23e89d1a5adb456cdd8da10dc06e24bd22c